client/src/App.js

```diff
@@ -1,20 +1,20 @@
 import React from "react";
-import { MuiThemeProvider } from "@material-ui/core";
-import { BrowserRouter, Route } from "react-router-dom";
 
-import { theme } from "./themes/theme";
-import LandingPage from "./pages/Landing";
+import { Dashboard } from './pages/Dashboard';
 
-import "./App.css";
 
-function App() {
-  return (
-    <MuiThemeProvider theme={theme}>
-      <BrowserRouter>
-        <Route path="/" component={LandingPage} />
-      </BrowserRouter>
-    </MuiThemeProvider>
-  );
+class App extends React.Component {
```

Comment:
does app have to be class? can it be a functional component instead?

Reply:
Yes functional one is better. I temporarily set this to a class one for a probable ref using.

Also if there's a lot of states, class one is easier to manage IMO.

Reply:
Have you ever tried useReducer hook for state management? useState is annoying when managing many states, but useReducer is good when you have a lot of states to manage i think. you might want to check it out if we are going to be using functional components.

Reply:
**Use useState if you have**

- a small application
- business logic is within your component
- simple state transitions

**Use useReducer if you have**

- complex state transitions
- complicated business logic more suitable for a reducer function
- need for a more predictable and maintainable state architecture
- the need to update state deep down in your component tree
- 

Since the Application we are working on is very small. So my recommendation is that we should go with useState

Reply:
hmm i think our application isn't that small though. I think we will be using global states in our app so useReducer will come in handy. Let's have a discussion on this in tomorrow's standup and get @bonnieli opinion on this as well

Reply:
@varunrampal @singonwan 
Thank you gys for your advice! Actually I am not good at Redux but I will learn it quickly if we need it.

I am still simplifying and abstracting my code for the dashboard, for both good look and convenience for the page routing

Reply:
I mix the Sidebar and the Header components as a single one since they use the same states for page rendering and routing. The cost is the file looks bigger and more complicated. You guys can check after I push it.

Reply:
For me it's acceptable, about 200 lines of code